integration/spark/src/main/common/java/io/openlineage/spark/agent/facets/builder/ErrorFacetBuilder.java

```diff
@@ -0,0 +1,26 @@
+package io.openlineage.spark.agent.facets.builder;
+
+import io.openlineage.client.OpenLineage.RunFacet;
+import io.openlineage.spark.agent.facets.ErrorFacet;
+import io.openlineage.spark.api.CustomFacetBuilder;
+import java.util.function.BiConsumer;
+import org.apache.spark.scheduler.JobFailed;
+import org.apache.spark.scheduler.SparkListenerJobEnd;
+
+/** {@link CustomFacetBuilder} that generates an {@link ErrorFacet} for job failure events. */
+public class ErrorFacetBuilder extends CustomFacetBuilder<SparkListenerJobEnd, RunFacet> {
```

Comment:
I really like that we now have consistent interface not only for datasets, but also for facets.